showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITEndpointContext.java

```diff
@@ -50,17 +130,72 @@ public void endpointResolution_default() throws InterruptedException, IOExceptio
 
   // User configuration
   @Test
-  public void endpointResolution_userConfiguration() throws InterruptedException, IOException {
+  public void endpointResolution_userSetEndpoint() throws InterruptedException, IOException {
     String customEndpoint = "test.com:123";
     EchoClient echoClient = null;
     try {
       EchoSettings echoSettings =
-          EchoSettings.newBuilder()
+          ExtendedEchoSettings.newBuilder()
+              .setCredentialsProvider(NoCredentialsProvider.create())
```

Comment:
I'm wondering if it's possible to override `getUniverseDomain()` in `NoCredentials`, so that we can test the universe domain validation as well.

Reply:
I ended up creating two new wrappers: `UniverseDomainCredentials` which wraps Credentials and `UniverseDomainCredentialsProvider` which wraps CredentialsProvider.

Showcase tests can set the universe domain to the Credentials so we can test the validation logic by checking that the universe domain matches.